onnxruntime/core/session/inference_session.cc

```diff
@@ -762,7 +762,7 @@ common::Status InferenceSession::InitializeSubgraphSessions(Graph& graph, Sessio
       // setup all the info for handling the feeds and fetches used in subgraph execution
       auto* p_op_kernel = session_state.GetMutableKernel(node.Index());
       ORT_ENFORCE(p_op_kernel);
-      auto& control_flow_kernel = dynamic_cast<controlflow::IControlFlowKernel&>(*p_op_kernel);
+      auto& control_flow_kernel = static_cast<controlflow::IControlFlowKernel&>(*p_op_kernel);
       ORT_RETURN_IF_ERROR_SESSIONID_(
```

Comment:
Would be good to add a comment saying this is safe as only control flow nodes have subgraphs (node.GetAttributeNameToMutableSubgraphMap() is empty for all other node types)

Reply:
Will do. But note that the dynamic_cast had to be safe already (it's casting to `controlflow::IControlFlowKernel&` instead of `controlflow::IControlFlowKernel*` - the latter returns null if the pointer can't be converted to IControlFlowKernel, while the former throws, since there are no null references in C++).